### ISSUE TITLE
robot extract fails with no explanation on some occasions if terms not found

### ISSUE THREAD

**Issue by cmungall** [ontodev/robot#285]
## Case 1, no CURIE expansion

foo.tsv:
```
X:1
X:2
```

foo.owl:
```
Ontology: <http://purl.obolibrary.org/obo/foo>
    
Class: <http://purl.obolibrary.org/obo/X_1>

    Annotations: 
        rdfs:label "foo"
```

extract:
```
$ robot extract -m TOP -T foo.tsv -i foo.owl -o z.owl
usage: robot extract --input <file> --term-file <file> --output <file>
             --output-iri <iri>
 -b,--branch-from-term <arg>    root term of branch to extract
 -B,--branch-from-terms <arg>   root terms of branches to extract
 -h,--help                      print usage information
 -i,--input <arg>               load ontology from a file
 -I,--input-iri <arg>           load ontology from an IRI
 -l,--lower-term <arg>          lower level term to extract
 -L,--lower-terms <arg>         lower level terms to extract
 -m,--method <arg>              extract method to use
 -noprefixes                    do not use default prefixes
 -o,--output <arg>              save ontology to a file
 -O,--output-iri <arg>          set OntologyIRI for output
 -p,--prefix <arg>              add a prefix 'foo: http://bar'
 -P,--prefixes <arg>            use prefixes from JSON-LD file
 -t,--term <arg>                term to extract
 -T,--term-file <arg>           load terms from a file
 -u,--upper-term <arg>          upper level term to extract
 -U,--upper-terms <arg>         upper level terms to extract
 -V,--version                   print version information
 -v,--verbose                   increased logging
 -vv,--very-verbose             high logging
 -vvv,--very-very-verbose       maximum logging, including stack traces
 -x,--xml-entities              use entity substitution with ontology XML
                                output
```

There is no indication of why this failed. It's presumably due to CURIE expansion failing for "X"

## Case 2, with CURIE expansion

```
GO:2
GO:3
```

```
Ontology: <http://purl.obolibrary.org/obo/foo>
    
Class: <http://purl.obolibrary.org/obo/GO_1>

    Annotations: 
        rdfs:label "foo"
```

same:
```
$ robot extract -m TOP -T foo.tsv -i foo.owl -o z.owl
usage: robot extract --input <file> --term-file <file> --output <file>
             --output-iri <iri>
```

However, if I remove *either* of the lines from the tsv I get

```
$ robot extract -m TOP -T foo.tsv -i foo.owl -o z.owl
EMPTY TERMS ERROR ontology does not contain input terms
For details see: http://robot.obolibrary.org/errors#empty-terms-error

usage: robot extract --input <file> --term-file <file> --output <file>
             --output-iri <iri>
```

**Comment by rctauber** [comment]
I ran into this the other day. I did something not very smart and it should be fixed by https://github.com/ontodev/robot/pull/283 ... if you test it out with that fix does it work?

**Comment by rctauber** [comment]
Fixed by #283
